File: PokeLangJava/src/Lexer/Comment.java
package Lexer;

import java.text.CharacterIterator;

public class Comment extends AFD {

    @Override
    public Token evaluate(CharacterIterator code, int line, int column) {
        String comment = "";

        if (code.current() == '#') {
            comment += "#";
            code.next();

            comment += readComment(code);

            if (endComment(code)) {
                return new Token("COMENTARIO", comment, line, column);
            }
        }

        return null;
    }

    private String readComment(CharacterIterator code) {
        String comment = "";

        while (code.current() != '\r' && code.current() != '\n' && code.current() != CharacterIterator.DONE) {
            comment += code.current();
            code.next();
        }

        return comment;
    }

    private boolean endComment(CharacterIterator code) {
        return code.current() == '\r' || code.current() == '\n' || code.current() == CharacterIterator.DONE;
    }
}

File: PokeLangPython/Lexer/Comment.py
from AFD import AFD
from Token import Token

class Comment(AFD):
    def evaluate(self, code, line, column):
        comment = ""

        for char in code:
            if char == '#':
                comment += "#"
                next_char = next(code, None)
                while next_char and next_char not in {'\r', '\n'}:
                    comment += next_char
                    next_char = next(code, None)

                if not next_char or next_char == '\r' or next_char == '\n':
                    return Token("COMENTARIO", comment, line, column)

        return None
